packages/spatial/src/renderer/components/FogSettingsComponent.test.tsx

```diff
@@ -0,0 +1,103 @@
+// provide for limited attribution for the Original Developer. In addition,
+// Exhibit A has been modified to be consistent with Exhibit B.
+
+// Software distributed under the License is distributed on an "AS IS" basis,
+// WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+// specific language governing rights and limitations under the License.
+
+// The Original Code is Ethereal Engine.
+
+// The Original Developer is the Initial Developer. The Initial Developer of the
+// Original Code is the Ethereal Engine team.
+
+// All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023
+// Ethereal Engine. All Rights Reserved.
+// */
+
+import {
+  Entity,
+  EntityUUID,
+  UUIDComponent,
+  createEntity,
+  destroyEngine,
+  getComponent,
+  getMutableComponent,
+  setComponent
+} from '@etherealengine/ecs'
+import { act, render } from '@testing-library/react'
+import assert from 'assert'
+import React from 'react'
+import { Fog, FogExp2, MathUtils } from 'three'
+import { CameraComponent } from '../../camera/components/CameraComponent'
+import { createEngine } from '../../initializeEngine'
+import { EntityTreeComponent } from '../../transform/components/EntityTree'
+import { RendererComponent } from '../WebGLRendererSystem'
+import { FogSettingsComponent, FogType } from './FogSettingsComponent'
+import { FogComponent, SceneComponent } from './SceneComponents'
+
+describe('FogSettingsComponent', () => {
+  let rootEntity: Entity
+  let entity: Entity
+
+  const mockCanvas = () => {
+    return {
+      getDrawingBufferSize: () => 0
+    } as any as HTMLCanvasElement
+  }
+
+  beforeEach(() => {
+    createEngine()
+
+    rootEntity = createEntity()
+    setComponent(rootEntity, UUIDComponent, MathUtils.generateUUID() as EntityUUID)
+    setComponent(rootEntity, EntityTreeComponent)
+    setComponent(rootEntity, CameraComponent)
+    setComponent(rootEntity, SceneComponent)
+    setComponent(rootEntity, RendererComponent, { canvas: mockCanvas() })
+
+    entity = createEntity()
+    setComponent(entity, UUIDComponent, MathUtils.generateUUID() as EntityUUID)
+    setComponent(entity, FogSettingsComponent)
+    setComponent(entity, EntityTreeComponent)
+
+    //set data to test
+    setComponent(rootEntity, SceneComponent, { children: [entity] })
+  })
+
+  afterEach(() => {
+    return destroyEngine()
+  })
+
+  it('Create Fog Setting Component', async () => {
+    const fogSettingsComponent = getMutableComponent(entity, FogSettingsComponent)
+    assert(fogSettingsComponent.value, 'fog setting component exists')
+
+    fogSettingsComponent.type.set(FogType.Height)
+    fogSettingsComponent.color.set('#ff0000')
+    fogSettingsComponent.far.set(2000)
+    fogSettingsComponent.near.set(2)
+    fogSettingsComponent.density.set(0.02)
+    const { rerender, unmount } = render(<></>)
+
+    await act(() => {
+      rerender(<></>)
+    })
```

Comment:
So after fixing fog with @dinomut1  and team there's one thing we should add into here and that is making sure the ShaderChunks from three.js are overrided. FogShader.ts I believe holds the functions that we use to override the shaders when brownian and height fog is selected. So we just need to assert that ShaderChunk.fog_fragment etc... are equal to our overrides. I can help with further context.